proposals/proposal_1/lib/src/items/item_list.dart

```diff
@@ -0,0 +1,139 @@
+import 'package:flutter/material.dart';
+import 'package:proposal_1/src/items/item.dart';
+import 'package:proposal_1/src/items/item_controller.dart';
+import 'package:proposal_1/src/items/item_details.dart';
+
+class ItemList extends StatefulWidget {
+  final ItemController controller;
+
+  const ItemList({Key key, @required this.controller}) : super(key: key);
+
+  @override
+  _ItemListState createState() => _ItemListState();
+}
+
+class _ItemListState extends State<ItemList> {
+  Item _selectedItem;
+
+  void _onItemTapped(Item item) {
+    setState(() {
+      _selectedItem = item;
+    });
+  }
+
+  @override
+  void initState() {
+    // Begin loading the users when the Item List is first shown.
+    widget.controller.initialize();
+
+    super.initState();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return AnimatedBuilder(
+      animation: widget.controller,
+      builder: (context, _) {
+        if (widget.controller.loading) {
+          return const Center(child: CircularProgressIndicator());
+        } else if (widget.controller.error) {
+          return const Center(child: Text('Oh no! Could not load the users'));
+        }
+
+        return LayoutBuilder(builder: (context, constraints) {
```

Comment:
You got it! After trying a number of different responsive designs, placing the LayoutBuilder over the whole Scaffold felt like the easiest option for folks to modify and extend. For example, the current design places an AppBar over the List and Item Details. However, many apps only display an AppBar above the item details instead.

Overall, I really wonder if we should include responsive breakpoints & the corresponding code. The more I played around with this,  the more I began to think that developers will end up deleting most of this and rewriting it depending on the design of their app.

  > Can we build the functionality in a way that 90% of developers will want to keep it and/or add to it, instead of ripping it out?

Where the LayoutBuilder needs to appear in the Widget hierarchy, which breakpoints they want to handle, etc all heavily depend to the unique design of their app and the goals for the organization. What do you think?

Reply:
That's fair. I'm glad we have the data now. Let's discuss.